src/node.rs

```diff
@@ -1 +1,148 @@
-pub struct Node;
+use conrod::{CustomWidget, CustomWidgetState};
+use conrod::{Color, Colorable};
+use conrod::Frameable;
+use conrod::{FontSize, Labelable};
+use conrod::{Mouse, MouseButtonState};
+use conrod::{Depth, Dimensions, Position, Positionable};
+use conrod::{UiId, Ui};
+use conrod::WidgetKind;
+
+#[derive(PartialEq, Copy, Clone, Debug)]
+pub enum NodeState {
+    Normal,
+    Dragging,
+}
+
+fn get_new_state(is_over: bool, prev: NodeState, mouse: Mouse) -> NodeState {
+    use self::NodeState::{Normal, Dragging};
+    match (is_over, prev, mouse.left) {
+        (true,  _,  MouseButtonState::Down) => Dragging,
+        _                                   => Normal,
+    }
+}
+
+impl NodeState {
+    /// Alter the widget color depending on the state.
+    fn color(&self, color: Color) -> Color {
+        match *self {
+            NodeState::Normal => color,
+            NodeState::Dragging => color.highlighted(),
+        }
+    }
+}
+
+impl CustomWidgetState for NodeState {
+    fn matches(&self, _other: &NodeState) -> bool { true }
+}
+
+#[derive(Clone, Debug)]
+pub struct Node<'a> {
+    pos: Position,
+    dim: Dimensions,
+    depth: Depth,
+    maybe_label: Option<&'a str>,
+    style: Style,
+}
+
+#[derive(PartialEq, Clone, Debug)]
+pub struct Style {
+    pub maybe_label_color: Option<Color>,
+    pub maybe_label_font_size: Option<u32>,
+}
+
+impl Style {
+    pub fn new() -> Style {
+        Style {
+            maybe_label_color: None,
+            maybe_label_font_size: None,
+        }
+    }
+}
+
+impl<'a> Node<'a> {
+    pub fn new() -> Node<'a> {
+        Node {
+            pos: Position::Absolute(0.0, 0.0),
+            dim: [64.0, 64.0],
+            depth: 0.0,
+            maybe_label: None,
+            style: Style::new(),
+        }
+    }
+}
+
+impl<'a> CustomWidget for Node<'a> {
+
+    type State = NodeState;
+
+    fn set<C>(self, ui_id: UiId, ui: &mut Ui<C>) {
```

Comment:
Rather than implementing the `set` method, implement the trait's `update` method (the `set` method now has a default implementation which shouldn't be overridden).